baselines/fedexp/fedexp/client.py

```diff
@@ -0,0 +1,98 @@
+"""Client for FedExp."""
+
+import copy
+from collections import OrderedDict
+from typing import Callable, Dict, List, Tuple
+
+import flwr as fl
+import torch
+from flwr.common import NDArrays, Scalar
+from hydra.utils import instantiate
+from omegaconf import DictConfig
+from torch.nn.utils import parameters_to_vector
+from torch.utils.data import DataLoader
+
+from fedexp.models import train
+
+
+class FlowerClient(fl.client.NumPyClient):
+    """Standard Flower client for CNN training."""
+
+    def __init__(
+        self,
+        cid: int,
+        net: torch.nn.Module,
+        train_loader: DataLoader,
+        device: str,
+        num_epochs: int,
+        data_ratio,
+    ):  # pylint: disable=too-many-arguments
+        print(f"Initializing Client {cid}")
+        self.cid = cid
+        self.net = net
+        self.train_loader = train_loader
+        self.device = torch.device(device)
```

Comment:
```suggestion
        self.device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
```

Reply:
If Flower's VirtualClientEngine is using GPUs (i.e. if `num_gpus>0`) then the clients will select device `cuda`, else `cpu`.